texar/data/data/multi_aligned_data.py

```diff
@@ -0,0 +1,701 @@
+                transforms = self._other_transforms[i]
+                if len(transforms) > 0:
+                    for transform in transforms:
+                        scalar_ex = transform(scalar_ex)
+                processed_examples.append(scalar_ex)
+            elif _is_record_data(dataset_hparams[i]["data_type"]):
+                example: Dict[str, Any] = raw_example_i  # type: ignore
+                # for some reasons even after the not None check here, mypy
+                # thinks self._dataset_features[i] may be None, hence adding
+                # ignore to the lines below
+                if self._dataset_features[i] is not None:
+                    for key, dtype in \
+                            self._dataset_features[i]["convert_types"].items():  # type: ignore # pylint: disable=line-too-long
+                        example[key] = np.asarray(example[key], dtype=dtype)
+                    for key, transform in \
+                            self._dataset_features[i]["image_transforms"].items():  # type: ignore # pylint: disable=line-too-long
+                        example[key] = transform(example[key])
+                for transform in self._other_transforms[i]:
+                    example = transform(example)
+                processed_examples.append(example)
+        return tuple(processed_examples)
+
+    def collate(self, examples) -> Batch:
+        dataset_hparams = self.hparams["datasets"]
+        transposed_examples: List[Tuple] = list(map(tuple, zip(*examples)))
+        batch: Dict[str, Any] = {}
+        for i, transposed_example in enumerate(transposed_examples):
+            if _is_text_data(dataset_hparams[i]["data_type"]):
+                # Treat it as MonoTextData and collate
+                pad_length = dataset_hparams[i]["max_seq_length"]
+                if pad_length is not None:
+                    pad_length += sum(x is not None and x != '' for x in
+                                      [self.vocab(i).bos_token,
+                                       self.vocab(i).eos_token])
+                text_ids = [self.vocab(i).map_tokens_to_ids_py(sent) for sent in
+                            transposed_example]
+                text_ids, lengths = padded_batch(
+                    text_ids, pad_length, pad_value=self.vocab(i).pad_token_id)
+                # also pad the examples
+                pad_length = pad_length or max(lengths)
+                transposed_example = tuple(
+                    sent + [''] * (pad_length - len(sent))
+                    if len(sent) < pad_length else sent
+                    for sent in transposed_example
+                )
+
+                text_ids = torch.from_numpy(text_ids).to(device=self.device)
+                lengths = torch.tensor(lengths, dtype=torch.long,
+                                       device=self.device)
+
+                batch.update(
+                    {connect_name(self._name_prefix[i], "text"):
```

Comment:
Can we preprocess these names? These names won't be changed and it's a bit of a waste to recompute it every time.